### PR TITLE
feat(Conductor)!: Vite conversion and necessary fixes

### PR DESCRIPTION
**Pull request by jakeaturner**

### CODE REVIEW

File: client/src/components/MaterialsManager/MaterialsManager.jsx [hunk 1]
```diff
@@ -381,9 +381,11 @@ const MaterialsManager = ({ projectID, show, onClose, ...props }) => {
 
   return (
     <Modal size="fullscreen" open={show} onClose={onClose} {...props}>
-      <Modal.Header>Manage Ancillary Materials</Modal.Header>
+      <Modal.Header>Manage Project Materials</Modal.Header>
```
Comment (ethanaturner) [inline client/src/components/MaterialsManager/MaterialsManager.jsx:384]: We probably shouldn't start shifting the UI for this until the backend implementation has been shifted towards Projects. Uploaded files will start to be listed in responses from the Books API.

File: client/src/Platform.jsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ import './styles/global.css';
  */
 const Platform = () => {
   /* Configure global Axios defaults */
-  axios.defaults.baseURL = '/api/v1';
+  axios.defaults.baseURL = `${import.meta.env.VITE_API_BASE_URL}/api/v1`;
```
Comment (ethanaturner) [inline client/src/Platform.jsx:19]: This should be conditionally included only during development. If left as an absolute URL, the client JS built at Docker build time will always point to LibreCommons, even if it is a Campus Commons instance.

**Review by ethanaturner — CHANGES_REQUESTED** [review]

File: client/src/components/MaterialsManager/MaterialsManager.jsx (hunk 1 above)
Comment (jakeaturner) [reply client/src/components/MaterialsManager/MaterialsManager.jsx:384]: Removed. This was WIP that wasn't intended for this commit

File: client/src/Platform.jsx (hunk 2 above)
Comment (jakeaturner) [reply client/src/Platform.jsx]: Fixed

File: client/package.json [hunk 3]
```diff
@@ -2,18 +2,19 @@
...
+    "serve": "vite preview"
   },
   "dependencies": {
     "@fortawesome/fontawesome-free": "^5.15.4",
     "@testing-library/jest-dom": "^5.14.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
+    "@types/node": "^18.14.0",
+    "@types/react": "^18.0.28",
+    "@types/react-dom": "^18.0.11",
```
Comment (ethanaturner) [inline client/package.json:17]: These should be moved to `devDependencies`. Not a blocker.

File: server/api.js [hunk 4]
```diff
@@ -39,13 +39,13 @@ router.use(cors({
   origin: function (origin, callback) {
     /* Build dynamic origins list */
     let allowedOrigins = [];
-    if (process.env.NODE_ENV === 'production') {
-      allowedOrigins = String(process.env.PRODUCTIONURLS).split(',');
+    if (import.meta.env.VITE_MODE === 'production') {
```
Comment (ethanaturner) [inline server/api.js:42]: This is server-side logic, Vite config can't be used here.

File: client/src/components/projects/ProjectView.jsx [hunk 5]
```diff
@@ -2235,14 +2235,14 @@ const ProjectView = (props) => {
...
-                          {(canViewDetails && project.hasCommonsBook) && (
+                          {(canViewDetails) && (
                             <div className="mt-1e">
                               <Button
                                 color="blue"
                                 compact
                                 onClick={handleOpenMaterialsModal}
                               >
-                                  Manage Ancillary Materials
+                                  Manage Project Materials
```
Comment (ethanaturner) [inline client/src/components/projects/ProjectView.jsx:2245]: Restore this file, UI transition is not ready yet.

**Review by ethanaturner — CHANGES_REQUESTED** [review]
Some small changes to make

**Review by ethanaturner — APPROVED** [review]
_ET_. 02/22/23.

**Pull request merged**